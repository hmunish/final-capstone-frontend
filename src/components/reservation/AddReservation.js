import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useSearchParams } from "react-router-dom";
import { getCars } from "../../redux/cars/carsSlice";

function AddReservation() {
  const { cars, isLoading } = useSelector((state) => state.cars);
  const dispatch = useDispatch();

  const [searchParams] = useSearchParams();
  const carId = searchParams.get("carId");

  useEffect(() => {
    dispatch(getCars());
  }, [dispatch, cars.length]);

  if (isLoading) {
    return (
      <div>
        <h1>Loading...</h1>
      </div>
    );
  }
  return (
    <section className="add-reservation">
      <div className="green-overlay" />
      <div className="wrapper">
        <h2>Book a car test-ride</h2>
        <hr />
        <p>
          There are 34 different models of our car. We have showrooms all over
          the globe which some include test riding facilities. If you wish to
          find out if a test ride is available in your area, then choose your
          car, bike, and location below.
        </p>
        <form className="reservation-form">
          <input type="date" name="bookingDate" />
          <select>
            <option default>Location</option>
            <option value="london">London</option>
            <option value="newyork">New York</option>
            <option value="capetown">Cape Town</option>
          </select>
          <select>
            <option default>Car Model</option>
            {cars.map((car) => (
              <option
                key={car.id}
                value={car.id}
                selected={+carId === car.id ? true : ""}
              >
                {car.name}
              </option>
            ))}
          </select>
          <input type="submit" value="Book Now" />
        </form>
      </div>
    </section>
  );
}

export default AddReservation;
