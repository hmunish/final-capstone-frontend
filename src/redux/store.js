import { configureStore } from '@reduxjs/toolkit';
import loginReducer from './login/loginSlice';
import signupReducer from './signup/signupSlice';

const store = configureStore({
  reducer: {
    login: loginReducer,
    signup: signupReducer,
  },
});

export default store;
