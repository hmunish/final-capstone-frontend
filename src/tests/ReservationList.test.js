import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import AddReservation from '../components/reservation/AddReservation';
import store from '../redux/store';

test('Car Details components Match', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <AddReservation />
      </Provider>
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});