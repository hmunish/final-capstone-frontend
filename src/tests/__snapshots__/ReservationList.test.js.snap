// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car Details components Match 1`] = `
<section
  className="add-reservation"
>
  <div
    className="green-overlay"
  />
  <div
    className="wrapper"
  >
    <h2>
      Book a car test-ride
    </h2>
    <hr />
    <p>
      There are 34 different models of our car. We have showrooms all over the globe which some include test riding facilities. If you wish to find out if a test ride is available in your area, then choose your car, bike, and location below.
    </p>
    <form
      className="reservation-form"
      onSubmit={[Function]}
    >
      <input
        name="bookingDate"
        required={true}
        type="date"
      />
      <select
        name="location"
        required={true}
      >
        <option
          default={true}
        >
          Location
        </option>
        <option
          value="london"
        >
          London
        </option>
        <option
          value="newyork"
        >
          New York
        </option>
        <option
          value="capetown"
        >
          Cape Town
        </option>
      </select>
      <select
        name="carId"
        required={true}
      >
        <option
          default={true}
        >
          Car Model
        </option>
      </select>
      <input
        type="submit"
        value="Book Now"
      />
    </form>
    <p />
    <p />
  </div>
</section>
`;
